### ISSUE TITLE
Cannot link my project against cpputest, always get undefined symbols for UtestShell::getProgressIndicator

### ISSUE THREAD

**Issue by spachner** [cpputest/cpputest#802]
Hi,
I installed current version og cpputest on my Mac running 10.11.1. Installation works file, but during linking with my project I get this undefined symbols.

Undefined symbols for architecture x86_64:
  "UtestShell::getProgressIndicator() const", referenced from:
      vtable for TEST_Basic_TypeTest_TestShell in testCmdBasics.o
      ...

Any hints?

Stefan

**Comment by basvodde** [comment]
Uhm, no. How did you build it? configure/make ?

**Comment by spachner** [comment]
Hi,

I installed according to manual:

cd cpputest-xxx/cpputest_build/
autoreconf .. -i
../configure
make
sudo make install

A couple of month ago this worked out, by using cpputest-3.7dev. I have still these sources and work on the lib made from that. But I would like to keep on track on the current versions. 

Stefan

**Comment by basvodde** [comment]
Ok, and you get the error at the "make" step? or in usage?

**Comment by spachner** [comment]
Hi,

making cpputest was without any problem. The "Undefined symbols" error appears while linking my project against cpputest.

Stefan

**Comment by basvodde** [comment]
Ah ok, so "testCmdBasics" is your test class? Did you try to clean (delete all .o files) and rebuild?

**Comment by arstrube** [comment]
In my experience, problems like that tend to occur when
- I forgot to specify `-lCppUTest` and `-lCppUTestExt`
- for some reason, include directory's and librares' version doesn't agree

All I can remember is I have seen the errors you see, and it was really because some stupid small oversight of mine (wish I could recall more, sorry).

Did you set your linker path (`-L`) to anything? if so, try removing that. On a Mac, it shouldn't be needed.

**Comment by spachner** [comment]
@basvodde: Yes, I tried a clean and rebuild. And I just installed freshly the 3.7.2. Problem persists:

Undefined symbols for architecture x86_64:
"UtestShell::getProgressIndicator() const"

@arstrube: my makefile must be correct otherwise It would not work with my "old" cpputest-3.7dev installation. And I just checked the headerfiles in /usr/local/include/CppUTest*. They changed with my last "sudo make install" of  3.7.2, so they must be one to be installed for 3.7.2.

I found that:

when having installed "cpputest-3.7dev":

<pre>
spachner$ find /usr/local/include/CppUTest* -exec grep getProgressIndicator {} \; -print
    virtual const char *getProgressIndicator() const;
    virtual const char* getProgressIndicator() const;
/usr/local/include/CppUTest/Utest.h
</pre>

when having installed "cpputest-3.7.2":

<pre>
spachner$ find /usr/local/include/CppUTest*/*.h -exec grep getProgressIndicator {} \; -print
***nothing output***
</pre>

Seems that getProgressIndicator() has been moved or deleted in Utest.h

**Comment by basvodde** [comment]
Indeed, the getProgressIndicator() is gone, which makes me wonder why you still have a linker reference to that one? Do you have any idea? Perhaps still a stale .o file? Or perhaps using the wrong Utest.h file?

It shouldn't have a reference to that method anymore...

**Comment by arstrube** [comment]
I am now fairly certain that your cpputest headers don't agree with your newly built library, for whatever reason. - you cannot install 3.7.2 and then link against int with 3.7dev. You have to specify both headers and library in your 3.7.dev path (NOT what's installed).

**Comment by arstrube** [comment]
I thin k you have to manually copy `cpputestbuild/lib/` to `lib/` as well.

**Comment by spachner** [comment]
Hi 

probelm is solved!

Both of you are was actually right. I included in my project not from standard include path rather then the 3.7dev includes which is a copy of the sources I added to my svn repository. This means that the under /usr/.. installed libs and includes were right, but my project was using the 3.7dev ones. 

Everything is my fault.

Many thanks for helping and sorry for stealing your time.

Have a nice day.

regards

Stefan
